Update margin-trim content

rohini-ranjanR opened the pull request:
Removing `in-flow` and `all` values issue #23012.

<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description
Removing `in-flow` and `all` values as it was not mentioned in https://w3c.github.io/csswg-drafts/css-box-4/#margin-trim
<!-- ✍️ Summarize your changes in one or two sentences -->

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
Fixes #23012 
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

rohini-ranjanR committed:
Update content

Removing inflow and all values issue #23012.

Josh-Cena commented:
We need the new property values here, like `block`, `inline`, etc. Right now there's only one value which doesn't make sense.

rohini-ranjanR commented:
Can I add these in place of `in-flow` and `all`

`block`

-  Computes to 'block-start block-end'

`inline`
- Computes to 'inline-start inline-end'

rohini-ranjanR committed:
Merge branch 'mdn:main' into main

rohini-ranjanR committed:
Adding block and inline

Adding `block` and `inline`

wbamberg commented:
> Can I add these in place of `in-flow` and `all`
> 
> `block`
> 
>     * Computes to 'block-start block-end'
> 
> 
> `inline`
> 
>     * Computes to 'inline-start inline-end'

Thank you for the update! We also need block-start, inline-start, inline-end, block-end.

You can see the alllowed values in the specification: https://w3c.github.io/csswg-drafts/css-box-4/#margin-trim.

dipikabh commented:
Hello @rohini-ranjanR, thank you for your contribution.

There are a couple of changes that would need to be done to really fix https://github.com/mdn/content/issues/23012:
- Remove the outdated values
- Document the new values. Ideally, the descriptions should be more explanatory than a copy&paste from the specification. 
- Update other impacted sections such as [Examples](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-trim#examples) to remove the outdated values and add code examples for the new values.

Please let us know if you would be interested in taking up this work.

rohini-ranjanR commented:
Hello, @dipikabh
Yes, I would like to work on this task.

dipikabh commented:
That sounds great, @rohini-ranjanR! You can push your changes/commits to this PR for this work.

Let me know if you have any questions or need any help with the documentation process.

rohini-ranjanR committed:
Update outdated content add new values

Update outdated content of inflow and all values and add content for block, inline, block-start, block-end, inline-start, inline-end, update the example and add one more example for the inline start. https://developer.mozilla.org/en-US/docs/Web/CSS/margin-trim#examples and https://w3c.github.io/csswg-drafts/css-box-4/#margin-trim used for the content.

dipikabh commented on files/en-us/web/css/margin-trim/index.md:
```diff
@@ -38,14 +42,25 @@ margin-trim: unset;
-  - : Trims the margins of in-flow boxes and floats whose margins coincide with the container's content edge.
+- `block-end`
+
+  - : Margins of last block child with the container's edge will be truncated to zero.
```
```suggestion
  - : Margin of last block child with the container's edge is trimmed to zero.
```

dipikabh commented on files/en-us/web/css/margin-trim/index.md:
```diff
@@ -38,14 +42,25 @@ margin-trim: unset;
+  - : Margins of last block child with the container's edge will be truncated to zero.
+
+- `inline`
+  - : Container truncate the margins provided to the inline children where they adjoin the container's edge, without affecting spacing at the beginning and end of the       row.
```
```suggestion
  - : Margins provided to the inline children where they adjoin the container's edges are trimmed to zero without affecting the spacing at the beginning and end of the row.
```

dipikabh commented on files/en-us/web/css/margin-trim/index.md:
```diff
@@ -38,14 +42,25 @@ margin-trim: unset;
+  - : Container truncate the margins provided to the inline children where they adjoin the container's edge, without affecting spacing at the beginning and end of the       row.
+
+- `inline-start`
+  - : Margins between the container's edge and the first inline child will be truncated to zero.
```
```suggestion
  - : Margin between the container's edge and the first inline child is trimmed to zero.
```

dipikabh commented on files/en-us/web/css/margin-trim/index.md:
```diff
@@ -38,14 +42,25 @@ margin-trim: unset;
+  - : Margins between the container's edge and the first inline child will be truncated to zero.
+
+- `inline-end`
+  - : Margin between the container's edge and the last inline child will be truncated to zero.
```
```suggestion
  - : Margin between the container's edge and the last inline child is trimmed to zero.
```

dipikabh commented on files/en-us/web/css/margin-trim/index.md:
````diff
@@ -85,14 +101,24 @@ The problem here is that you'd end up with 20px too much spacing at the right of
+}
+```
+
+Similarly, to remove left margins with the container's edge:
````
```suggestion
Similarly, to remove left margin with the container's edge:
```

dipikabh reviewed (changes requested):
Thanks @rohini-ranjanR! I've left some edit comments.
I just realized that these new values are not yet implemented in any browser and this documentation is being updated ahead of its time.

rohini-ranjanR committed:
Update files/en-us/web/css/margin-trim/index.md

Updated

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

rohini-ranjanR committed:
Update files/en-us/web/css/margin-trim/index.md

Updated

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

rohini-ranjanR committed:
Applying suggestions

rohini-ranjanR committed:
Suggestion applied

rohini-ranjanR committed:
Suggestion applied

rohini-ranjanR committed:
Update files/en-us/web/css/margin-trim/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

rohini-ranjanR committed:
Update files/en-us/web/css/margin-trim/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

rohini-ranjanR commented:
I have a doubt about the line 'The margin-trim property allows the container to trim the margins of its children where they adjoin the container's edges.' code line 20 do they should be updated to 'The margin-trim property allows to trim the margins of its children where they adjoin the container's edges. '

dipikabh reviewed (approved):
Thanks for the updates, @rohini-ranjanR!
